### PR TITLE
Enhanced the release-production workflow to support conditional checks to publish and user entered tags.

### PR DESCRIPTION
**Pull request by ebadiere**
Docker relelase build image fix. 

**Related issue(s)**:  #1898

Fixes #

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Dockerfile
-packages/relay/node_modules/
-packages/relay/tests/
-packages/server/node_modules/
-packages/server/tests/
-node_modules/
-tools/
\ No newline at end of file
+**/packages/relay/node_modules/
```
Comment (Nana-EC) [inline .dockerignore:2]: Which package dirs does this refer to?

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Unclear which dirs under the repo this refers to.
If the goal is when the process is run by an external process then please update the PR description to make this clear and if possible add a comment in the file

File: .dockerignore (hunk 1 above)
Comment (ebadiere) [reply .dockerignore:2]: I read this to mean the `node_modules` under the relay.  My approach was to modify what was already there to use the proper syntax.  The `packages/relay/node_modules/` was already listed in the .dockerignore file.

File: .github/workflows/release-production.yml [hunk 2]
```diff
@@ -33,14 +41,25 @@ jobs:
...
-      - name: Build and push images
+      # Build the Docker image
+      - name: Build Docker image
+        id: docker_build
         uses: docker/build-push-action@v2
         with:
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-          platforms: linux/amd64, linux/arm64
+          context: .
```
Comment (Nana-EC) [inline .github/workflows/release-production.yml:49]: Why these context changes

**Review by Nana-EC — CHANGES_REQUESTED** [review]
dispatch should be in additional, also not clear on the additional changes

File: .github/workflows/release-production.yml [hunk 3]
```diff
@@ -1,8 +1,16 @@
 name: Release Production Environment
 
 on:
-  push:
-    tags: [ v* ]
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (ebadiere) [reply .github/workflows/release-production.yml:4]: Added `push` back.

File: .github/workflows/release-production.yml [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
+    inputs:
+      should_push:
```
[the remark this answers is not in the document]
Comment (ebadiere) [reply .github/workflows/release-production.yml:6]: Updated.

File: .github/workflows/release-production.yml [hunk 5]
```diff
@@ -1,8 +1,16 @@
...
+        description: 'Set to true to push the image after building'
```
[the remark this answers is not in the document]
Comment (ebadiere) [reply .github/workflows/release-production.yml:7]: Updated.

File: .github/workflows/release-production.yml (hunk 2 above)
Comment (ebadiere) [reply .github/workflows/release-production.yml:49]: I missed the original context settings when refactoring.  They are restored.

File: .github/workflows/release-production.yml [hunk 6]
```diff
@@ -33,14 +41,25 @@ jobs:
...
+      # Push the Docker image
+      - name: Push Docker image
+        if: success() && github.event.inputs.should_push == 'true'
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
           push: true
-          tags: ${{ env.REGISTRY }}/${{ github.repository }}:${{env.TAG}}
+          tags: my-image:${{ github.event.inputs.image_tag }}
```
[the remark this answers is not in the document]
Comment (ebadiere) [reply .github/workflows/release-production.yml:62]: I was thinking of the custom tag.  I updated it to now define the old default tag and only use the custom tag if it has been provided.

File: .github/workflows/release-production.yml [hunk 7]
```diff
@@ -4,6 +4,17 @@ on:
   push:
     tags: [ v* ]
 
+  workflow_dispatch:
+    inputs:
+      publish:
```
Comment (Nana-EC) [inline .github/workflows/release-production.yml:9]: Input unused.
You should make it so only the build process runs if `publish` is false

File: .github/workflows/release-production.yml [hunk 8]
```diff
@@ -33,14 +51,29 @@ jobs:
...
+      - name: Build Docker image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          platforms: linux/amd64, linux/arm64          
+          file: ./Dockerfile
+          load: true # This loads the image into the Docker environment
+          tags: my-image:${{ env.IMAGE_TAG }}
```
Comment (Nana-EC) [inline .github/workflows/release-production.yml]: Fix tag value

File: .github/workflows/release-production.yml [hunk 9]
```diff
@@ -33,14 +51,29 @@ jobs:
...
+        if: success() && github.event.inputs.should_push == 'true'
         uses: docker/build-push-action@v2
         with:
           cache-from: type=gha
           cache-to: type=gha,mode=max
           platforms: linux/amd64, linux/arm64
+          file: ./Dockerfile
           push: true
-          tags: ${{ env.REGISTRY }}/${{ github.repository }}:${{env.TAG}}
+          tags: my-image:${{ env.IMAGE_TAG }}
```
Comment (Nana-EC) [inline .github/workflows/release-production.yml]: Why tag change?

File: .github/workflows/release-production.yml [hunk 10]
```diff
@@ -20,6 +31,13 @@ jobs:
       - name: Get tag
         run: echo "TAG=${GITHUB_REF#refs/*/v}" >> $GITHUB_ENV
 
+      - name: Set default tag
```
Comment (Nana-EC) [inline .github/workflows/release-production.yml:34]: This step isn't needed if it's a version tag. You should remove or else you'll break the release flow

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Still some changes left that break release flow

File: .github/workflows/release-production.yml [hunk 11]
```diff
@@ -20,6 +31,13 @@ jobs:
...
+        run: echo "IMAGE_TAG=${{ env.REGISTRY }}/${{ github.repository }}:${{ env.TAG }}" >> $GITHUB_ENV
+
+      - name: Override tag if custom tag is provided
+        if: github.event.inputs.image_tag != ''
+        run: echo "IMAGE_TAG=${{ github.event.inputs.image_tag }}" >> $GITHUB_ENV
```
Comment (Nana-EC) [inline .github/workflows/release-production.yml:39]: Easier if you use the same env variable
```suggestion
        run: echo "TAG=${{ github.event.inputs.image_tag }}" >> $GITHUB_ENV        
```

**Review by Nana-EC — CHANGES_REQUESTED** [review]

File: .github/workflows/release-production.yml (hunk 8 above)
Comment (ebadiere) [reply .github/workflows/release-production.yml]: my-image removed.

File: .github/workflows/release-production.yml (hunk 9 above)
Comment (ebadiere) [reply .github/workflows/release-production.yml]: my-image removed

File: .github/workflows/release-production.yml (hunk 10 above)
Comment (ebadiere) [reply .github/workflows/release-production.yml:34]: Updated.

File: .github/workflows/release-production.yml (hunk 11 above)
Comment (ebadiere) [reply .github/workflows/release-production.yml:39]: Updated.

File: .github/workflows/release-production.yml (hunk 7 above)
Comment (ebadiere) [reply .github/workflows/release-production.yml:9]: Updated.

**Review by Nana-EC — CHANGES_REQUESTED** [review]

**Pull request closed without merging**